djconnectwise/sync.py

```diff
@@ -1419,17 +1429,17 @@ def _assign_field_data(self, instance, json_data):
         ticket_class = models.Ticket
         activity_class = models.Activity
         try:
-            uid = json_data['objectId']
-        except KeyError:
+            uid = json_data.get('objectId')
+        except AttributeError:
             raise InvalidObjectException(
                 'Schedule entry {} has no objectId key to find its target'
                 '- skipping.'.format(instance.id)
             )
 
-        if json_data['type']['identifier'] == "S":
+        if json_data.get('type').get('identifier') == "S":
```

Comment:
Calling `.get()` on the return of `.get()` here is not a good idea. I think its save to use `json_data['type'].get()` here.